src/org/labkey/test/params/FieldDefinition.java

```diff
@@ -33,6 +34,11 @@
     private FieldValidator _validator;
     private String _url;
     private Integer _scale;
+    private Optional<Boolean> _hidden = Optional.empty();
+    private Optional<Boolean> _shownInDetailsView = Optional.empty();;
+    private Optional<Boolean> _shownInInsertView = Optional.empty();;
+    private Optional<Boolean> _shownInUpdateView = Optional.empty();;
+    private Optional<Boolean> _isPrimaryKey = Optional.empty();;
```

Comment:
Okay. I'll stop using Optional<T> in FieldDefinition.

I will patch off the start I've made to ListTest (in order not to throw away the work) because we still need it eventually.  It doesn't have to go in this commit